### ISSUE TITLE
Mod suggestion compilation 

### ISSUE THREAD

**Issue by Hyper-Tex** [MrJake222/AUNIS#36]
Since you want suggestions here too, I've compiled these suggestions of mine from the discord

1: SGC alarms.
These can be in the form of a block, or maybe a speaker model of some type. A link to the various sgc alarms is here https://www.youtube.com/watch?v=IF0lm-BM_9g the iconic ones are at 4:33 - 4:53 and 9:09 - 9:19 Alarms could be manually cycled through by right clicking, and activated via redstone signal, either manually, or if already implemented, from the gate itself, or for better control, using opencomputers. Since you were thinking of making a... walter-cube as you called it, you can integrate this too into that block.

2:  Other gate types/zpms/loot
If you add ZPMs, they should be rare loot in dungeon chests, Specifically Stronghold chests. in Vanilla, the rarest item in Stronghold loot is the Enchanted Golden Apple (or Notch apples if you prefer, which are uncraftable by the way) with a 2.5% chance of being in one, ZPMs should be rarer at... 1% - 1.5% also uncraftable to make it more like the show as they couldn't just make them. This would also make them VERY valuable, and more or less balanced so abuse would be minimized.
Of course there WAS an artificial ZPM like generator in the episode "the fifth race", but for this mod, it would need to have an EXTREME amount or resources to create and it could only be used once, as per the show... Okay twice, but only after repair, but that's it! (I would only have it as a one use item)
As for other stargates, that's more simple. since they are effectually the same, just different visually, just craft the stargate base block with say... a ring of blue wool or a ring of the blue crystals to make a pegasus gate, or a ring of black and white wool (four each) or a ring of ender and white crystals (again 4 each) to make a universe gate.
Also, if you do implement other gate types, the auto spawning has to be changed for the mysterious pages, for milky-way gate (or local galaxy gates... 7 symbol addresses) the spawn distance should be no more than... 70,000 blocks away? For Pegasus gates (or non-local galaxy gates, 8 symbol addresses) the range to spawn should be 100,000 to 250,000 blocks away. And lastly, the universe gates (or just 9 symbol addresses) should spawn at least 3 million to 5 million blocks away. (Extreme? Yes, that's the point XD) That's just for the overworld. For a final thing with stargates, where as the milky-way and Pegasus gates would be able to access the nether, just the universe gates can access The End, or dimensions from other mods. Heck, make a config that would prevent vanilla end portals from spawning, making universe gates the only thing capable of reaching The End. Hold up, now that I think about it, you could spawn a chest in where the end portal is/was containing a page with a special 9 symbol address only able to access The End, kinda like the Destiny address was the only 9 symbol address able to reach it. 
Going back to the ZPMs a moment, when powering the gate for 8/9 symbol addresses. When dialing an 8 symbol address, only one ZPM is required to dial. 5% of the ZPM will drain instantly, and will drain by 1% per minute active. And for 9 symbol addresses, oh boy, 3 Zpms required, will drain 40% from each ZPM instantly, and will have a 1% drain every 5 seconds! Of course this can have a config to adjust that.

3: random offworld activations/raids (should implement only after irises are part of the mod)
The idea here is that the gate will activate and hostile mobs will come through about 10 seconds later after the wormhole is established in a squad of about 10 to 20. Of course, having them appear at any random gate at any time could cause lag, so I was thinking of the idea of claimed gates. Now, the only way I can think of to implement that, would be to have some sort of... registration crystal of some sort, or using opencomputers maybe. Claimed gates would also be the only ones that could have irises/energy shields installed on them. Now, that doesn't mean non claimed gates can't have random offworld activations too, but should happen with extreme low frequency. Also more rarely on occasion, you could have a friendly mob or no mobs come through on these activations, and most rarely, have items come through, with the rarest being a ZPM naturally. That way it adds a risk/reward system to using an iris or not. Also, GDOs for irises/gate energy shields. 

4: weapons
Just the more iconic ones really, like the Jaffa staffs, Zat guns. the ori soldier staffs, Asuran (human replicators of Pegasus) blasters, Ronon's gun, and of course, the P90. Oh the Goa'uld ribbon device too. Could also be used at a remote ring teleporter control.

5: wormhole jumping
I don't know if this would be possible, but in the show a large enough explosion at a safe range from the gate, or a smaller directed explosion at the gate would cause the wormhole to jump to a nearby gate, usually the closest one to the receiving gate, provided the explosion doesn't destroy the gate outright. Energy weapons can cause the jump too, if hit enough times and fast enough.

6: DHD visuals
On receiving gates (the ones you actually come out from) the symbols on the DHD are not supposed to light up, as it's on a receiving gate. The big red/blue button too, but I guess that could stay lit to show the gate's on, and without the symbols, could be a subtle way to say DO NOT ENTER THIS SIDE, YOU WILL DIE, AS THIS END LEADS NOWHERE! XD

That's it for the ideas I suggested on Discord, hopefully some of these will make it ^_^

**Comment by MrJake222** [comment]
Wow, thanks for such a big and detailed description. Will definetly use it, someday...

**Comment by Hyper-Tex** [comment]
Of course. *nods* It only make sense to describe how these suggestions should work. Being a long time fan of the SG franchise, any mod that brings this series into it is something I wanna support greatly, and to get it as close to the series as possible, while staying within minecraft's limitations.

**Comment by MrJake222** [comment]
In my opinion, changes should follow as shown:

6. DHD visuals (pretty easy to do)
2. Other gates (Orlin in alpha testing, Universe model ready, Pegasus not yet)
     Rework of a power system (standalone, naquadah generator, ZPMs and changes you described)
1. Alarms (Not so crucial as above and not so-much-requested I think)
3. Raids (Should be pretty easy, just make the gate appear as open and spawn some zombies or other (custom?) mobs. Maybe claimed would mean they made a outbound connection recently? And they could leave a note with their homeworld address...)
4. Weapons (hard to do IMO, no models, might be OP. I'd prefer to do more "Stargate'ish" ones, like staff weapon and zats)
5. Jumping wormhole (Definetly possible, BUT current format of saving gates has no way of finding nearest gates without scanning through ALL gates in dimension. More technically, the keys in a Java's map are addresses, not positions)

**Comment by ShicKla** [comment]
Out of curiosity, will the Orlin gate be single use like it was in the show?

**Comment by Hyper-Tex** [comment]
Well, like I said in discord, these suggestions are up to you to implement, I'm just suggesting stuff.
I still stand firm of how number 3 should work though, again though, it is your mod, so do as you will. As for the mobs, not just Zombies. Creepers, Skeletons, and Spiders too naturally, especially at night, and if you ever wanna to a custom mob someday, like making a jaffa mob, that could be used too.
As for number 2, that was more a suggestion for WAY into the future, once this mod has developed quite a bit more
And ShicKla makes a good point about the Orlin gate. As far as we know, that was a single use gate, it'd be kinda silly if it took too many resources to make

**Comment by Phoenix10020** [comment]
Also adding a config option for the inner rings rotation and sound due to that being only for dialing the gate from the SGC, where when dialing with the DHD the inner ring doesn't spin?

**Comment by Cronosus** [comment]
about gates.. what about Tollan gate? the one they builded themselves after they escaped to planet without one

and i would be also glad for weapons

**Comment by catalyticenzyme** [comment]
> For a final thing with stargates, where as the milky-way and Pegasus gates would be able to access the nether, just the universe gates can access The End, or dimensions from other mods. Heck, make a config that would prevent vanilla end portals from spawning, making universe gates the only thing capable of reaching The End. Hold up, now that I think about it, you could spawn a chest in where the end portal is/was containing a page with a special 9 symbol address only able to access The End, kinda like the Destiny address was the only 9 symbol address able to reach it.

What happens if a prankster/griefer destroys the gate that that 9 symbol address goes to? What if the dragon destroys it? Is everyone locked out of the End forever (unless they know the address of a gate that was built elsewhere in the End)? In addition, this ruins what's left of the balance of the End fight—in vanilla you and the dragon battle to the death with no escape, and while currently you *can* get out by gating out of the End, there's no convenient premade gate and you have to have the foresight to bring your own gate materials, which puts a little bit of balance on it.
